src/Swashbuckle.AspNetCore.Filters/Examples/ResponseExample.cs

```diff
@@ -36,27 +35,61 @@ internal class ResponseExample
             }
 
             var response = operation.Responses.FirstOrDefault(r => r.Key == statusCode.ToString());
+            if (response.Equals(default(KeyValuePair<string, OpenApiResponse>)) != false || response.Value == null)
+            {
+                return;
+            }
 
-            if (response.Equals(default(KeyValuePair<string, OpenApiResponse>)) == false && response.Value != null)
+            var serializerSettings = serializerSettingsDuplicator.SerializerSettings(contractResolver, jsonConverter);
+            var multiple = example as IEnumerable<ISwaggerExample<object>>;
+            if (multiple == null)
             {
-                var serializerSettings = serializerSettingsDuplicator.SerializerSettings(contractResolver, jsonConverter);
-                var jsonExample = new OpenApiRawString(jsonFormatter.FormatJson(example, serializerSettings));
+                var jsonExample = new Lazy<IOpenApiAny>(() => new OpenApiRawString(jsonFormatter.FormatJson(example, serializerSettings)));
```

Comment:
I've never used Lazy<T> before... just wondering why you're using it?

Reply:
I saw that you were first looping through the request or response media types once (with .Any()) to determine whether an XML example would be required in order to create one, then looping though them a second time to add the examples.  You were also always creating a JSON example even though theoretically, it might not be used (if there is no JSON media type).

By using Lazy<>, the XML and JSON examples are each created when they are first used, and if one or both aren't used, then you don't pay the cost of creating them.  It allows you loop through the media types only once and simplifies the code.

Reply:
Ah OK, thanks for the explanation. I've never used Lazy<> before, and I now understand why it can be used. I'm not sure that there's much savings to be made in this use case though.

I'm on holiday without Visual Studio and am looking at your PR in the browser, I'll have a proper look when I get a chance.